idunn/api/geocoder.py

```diff
@@ -51,10 +52,12 @@ async def get_intentions():
         bragi_client.autocomplete(query, extra), get_intentions()
     )
     if intentions is not None and query.nlu:
-        intentions = list(
-            filter(
-                lambda i: post_filter_intention(i, autocomplete_response, query.limit), intentions
+        if autocomplete_nlu_filter_intentions:
+            intentions = list(
+                filter(
+                    lambda i: post_filter_intention(i, autocomplete_response, query.limit),
+                    intentions,
+                )
```

Comment:
I'm never sure what's more readable between list comprehension and `list(filter(...))` + lambda. Python has just too many reasonable alternatives :p

```suggestion
        intentions = [
            i
            for i in intentions
            if post_filter_intention(i, autocomplete_response, query.limit)
        ]
```

Not very important, just something I've had on my mind recently...

Reply:
I should definitely use list comprehension more systematically. Less natural to write in my opinion, but more readable probably.